src/Entity/Filesystem/File.php

```diff
@@ -0,0 +1,397 @@
+     * @Assert\Length(max=100)
+     *
+     * @SymfonySerializer\Groups({"autocomplete"})
+     */
+    private $name;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=20)
+     *
+     * @Assert\NotBlank
+     * @Assert\Length(max=20)
+     * @Assert\Choice(
+     *     callback={"App\Entity\Filesystem\FileTypeEnum", "toArray"},
+     *     strict=true
+     * )
+     */
+    private $type = FileTypeEnum::FILE;
+
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\File(
+     *     maxSize="5M",
+     *     mimeTypes={
+     *         "image/*",
+     *         "video/mpeg",
+     *         "video/mp4",
+     *         "video/quicktime",
+     *         "video/webm",
+     *         "application/pdf",
+     *         "application/x-pdf",
+     *         "application/vnd.ms-powerpoint",
+     *         "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+     *         "application/msword",
+     *         "application/vnd.ms-excel",
+     *         "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+     *         "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+     *         "application/rtf",
+     *         "text/plain",
+     *         "text/csv",
+     *         "text/html",
+     *         "text/calendar"
+     *     }
+     * )
+     */
+    private $file;
+
+    /**
+     * @var Administrator|null
+     *
+     * @Gedmo\Blameable(on="create")
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Administrator")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private $createdBy;
+
+    /**
+     * @var Administrator|null
+     *
+     * @Gedmo\Blameable(on="update")
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Administrator")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private $updatedBy;
+
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default": true})
+     */
+    private $displayed = true;
+
+    /**
+     * @var FilePermission[]|Collection
+     *
+     * @ORM\OneToMany(targetEntity="App\Entity\Filesystem\FilePermission", mappedBy="file", cascade={"all"}, orphanRemoval=true)
+     *
+     * @Assert\Valid
+     */
+    private $permissions;
+
+    /**
+     * @var File|null
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Filesystem\File", cascade={"persist"})
+     * @ORM\JoinColumn(onDelete="CASCADE", nullable=true)
+     *
+     * @Assert\Valid
+     */
+    private $parent;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=255, nullable=true)
+     *
+     * @Assert\Length(max=255, maxMessage="file.validation.filename_length")
+     */
+    private $originalFilename;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=10, nullable=true)
+     *
+     * @Assert\Length(max=10)
+     */
+    private $extension;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=50, nullable=true)
```

Comment:
`application/vnd.openxmlformats-officedocument.spreadsheetml.sheet` takes 66 chars, IMO we can remove these limites

Reply:
Nice catch!
It's better to increase the column size.